<?xml version="1.0"?>


<launch>

  <arg name="start_manager" default="true"/>
  <arg name="nodelet_manager" default="registration_manager"/>
  <!-- start nodelet manager -->
  <node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>

   <node pkg="nodelet" type="nodelet" name="pass_through" args="load pcl/PassThrough $(arg nodelet_manager)" output="screen">
    <remap from="~input" to="/cam3d/rgb/points" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.0
      filter_limit_max: 5.0
      keep_organized: true
    </rosparam>
  </node>

  <node pkg="cob_keyframe_detector" type="keyframe_detector_node" name="keyframe_detector_node" output="screen">
    <remap from="camera_info" to="/cam3d/rgb/camera_info"/>

    <param name="r_limit_" type="double" value="0.013" />
    <param name="p_limit_" type="double" value="0.013" />
    <param name="y_limit" type="double" value="0.013" />
    <param name="distance_limit_" type="double" value="0.025"/>
    <param name="trigger_always" type="bool" value="false"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="registration_nodelet" args="load cob_3d_registration/RegistrationNodelet $(arg nodelet_manager)" output="screen">

    <remap from="camera_info" to="/cam3d/rgb/camera_info"/>
    <remap from="point_cloud2" to="/pass_through/output"/>
    <remap from="point_cloud2_aligned" to="registration/pc_aligned"/>

      <param name="algo" value="info" />
      <param name="threshold_step" value="0.06" />
      <param name="threshold_diff" value="0.06" />
      <param name="min_info" value="1" />
      <param name="max_info" value="17" />
      <param name="always_relevant_changes" value="1" />

  </node>

</launch>
